stanford_soe_r25.pages.inc

```diff
@@ -0,0 +1,994 @@
+    $max_hours = 2;
+    if (!empty($rooms[$room]['max_hours'])) {
+      $max_hours = intval($rooms[$room]['max_hours']);
+      if ($max_hours == 0) {
+        $max_hours = 24;
+      }
+    }
+    $hours_array = array();
+    if ($max_hours > 2) {
+      for ($i = 0; $i < $max_hours; $i++) {
+        $hstr = '';
+        if ($i == 0) {
+          $hours_array[] = '30 minutes';
+          $hours_array[] = '1 hour';
+        }
+        else {
+          $hours_array[] = strval($i) . '.5 hours';
+          $hours_array[] = strval($i + 1) . ' hours';
+        }
+      }
+    }
+    else {
+      $hours_array[0] = '30 minutes';
+      $hours_array[1] = '60 minutes';
+      if ($max_hours > 1) {
+        $hours_array[2] = '90 minutes';
+        $hours_array[3] = '120 minutes';
+      }
+    }
+    $form['stanford_soe_r25_booking_duration'] = array(
+      '#type' => 'select',
+      '#title' => t('Duration'),
+      '#options' => $hours_array,
+      '#default_value' => 0,
+      '#required' => TRUE,
+    );
+  }
+  else {
+    // multi-day rooms have an end date and time instead of duration
+    $max_hours = '';
+    $form['stanford_soe_r25_booking_enddate'] = array(
+      '#type' => 'date_popup',
+      '#default_value' => format_date(time(), 'custom', 'Y-m-d H:i:s'),
+      '#date_type' => DATE_DATETIME,
+      '#date_timezone' => date_default_timezone(),
+      '#date_format' => 'Y-m-d h:i a',
+      '#date_increment' => 30,
+      '#date_year_range' => '-0:+1',
+      '#required' => TRUE,
+      '#title' => 'End Date/Time',
+    );
+  }
+  // max headcount for a room comes from parameter passed to the function
+  $form['stanford_soe_r25_booking_headcount'] = array(
+    '#type' => 'select',
+    '#title' => t('Headcount'),
+    '#options' => array(),
+    '#required' => TRUE,
+  );
+  // add to the select list for the number of possible headcounts
+  for ($i = 1; $i < $max_headcount + 1; $i++) {
+    $form['stanford_soe_r25_booking_headcount']['#options'][] = strval($i);
+  }
+  // every booking needs some reason text
+  $form['stanford_soe_r25_booking_reason'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Reason'),
+    '#required' => TRUE,
+    '#maxlength' => 40,
+  );
+
+  // check for event attribute fields, and build 'em
+  // each of these corresponds to a "custom attribute" for events in 25Live
+  // and are specified in our room config as an array of attrib ids, field name,
+  // and field type
+  if (!empty($rooms[$room]['event_attribute_fields'])) {
+    foreach ($rooms[$room]['event_attribute_fields'] as $attr_id => $attr_info) {
+      switch ($attr_info['type']) {
+        case 'S':
+          $field_type = 'textfield';
+          break;
+        case 'B':
+          $field_type = 'checkbox';
+          break;
+        case 'X':
+          $field_type = 'textarea';
+          break;
+        default:
+          $field_type = '';
+      }
+      if (module_exists('stanford_pta') && !empty($attr_info['pta'])) {
```

Comment:
What is the stanford_pta module?

Reply:
https://code.stanford.edu/ksharp/stanford_pta